src/ring_theory/class_group.lean

```diff
@@ -99,6 +99,55 @@ noncomputable def class_group.mk : (fractional_ideal R⁰ K)ˣ →* class_group
 (quotient_group.mk' (to_principal_ideal R (fraction_ring R)).range).comp
   (units.map (fractional_ideal.canonical_equiv R⁰ K (fraction_ring R)))
 
+lemma class_group.mk_eq_mk {I J : (fractional_ideal R⁰ $ fraction_ring R)ˣ} :
+  class_group.mk I = class_group.mk J
+    ↔ ∃ x : (fraction_ring R)ˣ, I * to_principal_ideal R (fraction_ring R) x = J :=
+begin
+  erw [quotient_group.mk'_eq_mk', canonical_equiv_self, units.map_id],
+  exact ⟨λ ⟨_, ⟨x, hx⟩, h⟩, ⟨x, hx.symm ▸ h⟩, λ ⟨_, h⟩, ⟨_, ⟨_, rfl⟩, h⟩⟩
+end
```

Comment:
There's a lemma for your `exact`:
```suggestion
by erw [quotient_group.mk'_eq_mk', canonical_equiv_self, units.map_id, set.exists_range_iff]; refl
```

Reply:
The style guide says:
```
If you are using multiple tactics inside a by ... block, use braces by { tac1, tac2 } rather than abusing the ; operator by tac1; tac2, which should only be used when multiple goals need to be processed by tac2.
```
So the `refl` unfortunately goes over to the next line.